openhands/server/app.py

```diff
@@ -48,6 +49,7 @@ async def health():
 app.include_router(security_api_router)
 app.include_router(feedback_api_router)
 
+AttachSessionMiddleware = import_from(config.attach_session_middleware_class)
```

Comment:
There's a lot to like about dynamically loading stuff (I'm guilty of a love for metaclasses even), but this bit is a little confusing to me. I'm concerned that it makes the code less clear, it's less understandable at a glance what is running and how come. The other PR (on refactoring github auth) adds this kind of imports to more places, too. The code seems to get quite a number of 'if this then load the other'.

Could we implement this via an interface or default class, where github auth is plugged in a single place perhaps?

Reply:
Agreed - a base class makes sense for this.